### PR TITLE
Remove superfluous flow control

### PR DESCRIPTION
**Pull request by davide-baldo**
- hide the inlet address from the other side in kafka workers
- removed superfluous flow control

Draft because i want to do some extra validation on the `MultiAddr` instantiation

### CODE REVIEW

**Review by SanjoDeundiak — COMMENTED** [review]
Love the direction of this PR!

**Comment by davide-baldo** [comment]
double-checked!

File: implementations/rust/ockam/ockam_api/src/nodes/service.rs [hunk 1]
```diff
@@ -491,9 +503,9 @@ impl NodeManager {
 
         debug!("connected to {connection_instance:?}");
 
-        // Every piece of the chain must be part of the session to allow communication
-        for service in intermediary_services {
-            connection_instance.add_consumer(&service);
+        // make sure the service before flow control is a consumer
+        if let Some(address) = consumer_worker {
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_api/src/nodes/service.rs:507]: Are there any use cases for that? I had an impression that we can just get rid of it

Comment (davide-baldo) [reply implementations/rust/ockam/ockam_api/src/nodes/service.rs:507]: In practice, I think It wouldn't break anything (other than `hop`), but I reckon that if the user provides an extra address in the `MultiAddr` we should do anything possible to make sure it's supported out of the box.

**Review by etorreborre — APPROVED** [review]

**Pull request merged**